### PR TITLE
docs: Add documentation built with TypeDoc

### PR DESCRIPTION
**Pull request by thundron**
Depends-on: #6 
Change-type: patch
Signed-off-by: Lorenzo Alberto Maria Ambrosi <lorenzothunder.ambrosi@gmail.com>

### CODE REVIEW

**Comment by jhermsmeier** [comment]
Is this still WIP?

**Comment by thundron** [comment]
Nope, good to go (aka be reviewed)

File: README.md [hunk 1]
```diff
@@ -42,6 +46,225 @@ nm.init()
...
 ```
 
-## References
+## HTTP APIs
+
+### Connect to network
+
+POST `/connect-network`
+
+Body:
```
Comment (jhermsmeier) [inline README.md]: Should this also mention the post data encoding?

File: README.md [hunk 2]
```diff
@@ -42,6 +46,225 @@ nm.init()
...
+}
+```
+
+### Toggle WiFi connection
+
+POST `/toggle-wifi`
+
+```JSON
+{
+  "value": true
```
Comment (jhermsmeier) [inline README.md]: Missing description of what this data is here – is this the request or the response body?

File: README.md [hunk 3]
```diff
@@ -42,6 +46,225 @@ nm.init()
...
+```HTTP
+HTTP/1.1 500 Internal Server Error
+{
+  "code": 500,
+  "message": "Could not forgetNetwork",
+  "data": {...}
+}
+```
+
+### Toggle WiFi connection
```
Comment (jhermsmeier) [inline README.md]: Does a call to this actually toggle a connection without supply of a value? What connection is being "toggled"?

File: README.md [hunk 4]
```diff
@@ -42,6 +46,225 @@ nm.init()
...
+  "data": {...}
+}
+```
+
+### Forget network
+
+POST `/forget-network`
+
+```JSON
+{
```
Comment (jhermsmeier) [inline README.md]: Is this request or response body?

**Review by jhermsmeier — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (thundron) [reply README.md]: The first connection available will be activated. This is the only way I found, so far, to re-activate WiFi, but I agree this doesn't look completely right

File: README.md (hunk 1 above)
Comment (thundron) [reply README.md]: Good call

File: README.md (hunk 3 above)
Comment (thundron) [reply README.md]: I should rename it to just Toggle WiFi, to make it more clear

Comment (jhermsmeier) [reply README.md]: How about "Enable WiFi"? Toggle suggests implicitly switching between "on" and "off" states to me (might be just me though)

File: README.md [hunk 5]
```diff
@@ -42,6 +46,229 @@ nm.init()
...
+  "message": "Could not forgetNetwork",
+  "data": {...}
+}
+```
+
+### Toggle WiFi
+
+POST `/toggle-wifi`
+
+Body (JSON):
```
Comment (jhermsmeier) [inline README.md]: What I meant here was actually which post data encoding is supposed to be used; i.e. `application/x-www-form-urlencoded` or `multipart/form-data`

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,465 @@
...
+```
+
+HTTP APIs
+---------
+
+### Connect to network
+
+POST `/connect-network`
+
+Body:
```
Comment (jhermsmeier) [inline docs/README.md]: Same here ;P I reckon, best would be to define it once, and reference it throughout the docs

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,465 @@
...
+  "message": "Could not getCurrentNetwork",
+  "data": {...}
+}
+```
+
+### Forget network
+
+POST `/forget-network`
+
+```JSON
```
Comment (jhermsmeier) [inline docs/README.md]: This still ain't giving any info about whether this is request or response data

File: README.md (hunk 3 above)
Comment (thundron) [reply README.md]: That's actually what it does, it turns WiFi on and off by the `value` Body param

File: README.md (hunk 5 above)
Comment (thundron) [reply README.md]: It's meant to hint `application/json`, but maybe I should make it more clear (following comment too) by specifying headers

File: docs/README.md (hunk 7 above)
Comment (thundron) [reply docs/README.md]: It's taken from the main README.md, so it's best to fix that before re-building docs (which use root README.md)

**Review by jhermsmeier — APPROVED** [review]

**Review by hedss — APPROVED** [review]

**Pull request merged**